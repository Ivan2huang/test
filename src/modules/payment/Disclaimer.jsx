import React from 'react';
import { makeStyles } from '@material-ui/core';
import { useIntl, FormattedMessage } from 'react-intl';
import Typography from '../../uiComponents/Typography';
import { formatMessage } from '../../helpers/helpers';
import CONFIG from '../../constants/config';
import { OPEN_TERM_OF_USE_MESSAGE_TYPE } from './constants';

const useStyles = makeStyles(theme => ({
  header: {
    marginBottom: theme.spacingX(2),
  },
  link: {
    color: theme.outline,
    cursor: 'pointer',
  },
}));

const Disclaimer = () => {
  const classes = useStyles();
  const intl = useIntl();

  const openTermOfUse = () => {
    if (window.ReactNativeWebView && window.ReactNativeWebView.postMessage) {
      const messageData = {
        type: OPEN_TERM_OF_USE_MESSAGE_TYPE,
        url: CONFIG.paymentTermOfUseUrl,
      };
      window.ReactNativeWebView.postMessage(JSON.stringify(messageData));
    } else {
      window.open(CONFIG.paymentTermOfUseUrl);
    }
  };

  return (
    <>
      <Typography type="style-8" fontWeight="bold" className={classes.header}>
        {formatMessage(
          intl,
          'payment.checkout.disclaimer.header',
          'Disclaimer',
        )}
      </Typography>
      <Typography type="style-8">
        <FormattedMessage
          id="payment.checkout.disclaimer.content"
          defaultMessage={`By purchasing, using or obtaining any products, services or
        consultations made available through HealthPass (a) you acknowledge and
        understand that HealthPass is developed, operated and maintained jointly by
        various third party providers; and (b) you agree that you have read and agree to
        the {termOfUseLink} of HealthPass ("Terms of Use") governing your usage of HealthPass, which
        provide amongst other matters, that all products, services and
        consultations made available through HealthPass are not provided by OCBC
        but by Third Party Providers (as defined under the Terms of Use) under
        an independent contract entered between you and the Third Party
        Provider, for which OCBC has no responsibility.`}
          values={{
            termOfUseLink: (
              <Typography
                type="style-8"
                component="span"
                className={classes.link}
                color="outline"
                onClick={openTermOfUse}
                data-testid="open-term-of-use"
              >
                {formatMessage(
                  intl,
                  'payment.checkout.disclaimer.termOfUse',
                  'Term of Use',
                )}
              </Typography>
            ),
          }}
        />
      </Typography>
    </>
  );
};

export default Disclaimer;
