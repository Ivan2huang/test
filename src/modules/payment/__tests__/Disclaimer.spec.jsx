import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import withIntl from '../../../i18n/withIntlProvider';
import withTheme from '../../../themes/withThemeProvider';
import Disclaimer from '../Disclaimer';

jest.mock(
  '../../../uiComponents/Typography',
  // eslint-disable-next-line react/prop-types
  () => ({ children, ...props }) => {
    return <div {...props}>{children}</div>;
  },
);

jest.mock(
  '../../../constants/config',
  // eslint-disable-next-line react/prop-types
  () => ({
    ...jest.requireActual('../../../constants/config'),
    paymentTermOfUseUrl: 'paymentTermOfUseUrl',
    defaultLanguage: 'en-HK',
  }),
);

jest.mock('../../../helpers/helpers', () => ({
  formatMessage: jest.fn((intl, id, defaultMessage) => defaultMessage),
}));

describe('Disclaimer Component', () => {
  let Component;

  window.open = jest.fn();

  beforeEach(() => {
    Component = withIntl(withTheme(Disclaimer));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    const { container } = render(<Component />);

    expect(container).toMatchSnapshot();
  });

  it('should call window.open', () => {
    const { getByTestId } = render(<Component />);

    const link = getByTestId('open-term-of-use');
    fireEvent.click(link);

    expect(window.open).toHaveBeenCalledWith('paymentTermOfUseUrl');
  });

  it('should post message to webview', () => {
    window.ReactNativeWebView = {
      postMessage: jest.fn(),
    };
    const { getByTestId } = render(<Component />);

    const link = getByTestId('open-term-of-use');
    fireEvent.click(link);

    expect(window.ReactNativeWebView.postMessage).toHaveBeenCalledWith(
      JSON.stringify({
        type: 'OPEN_TERM_OF_USE',
        url: 'paymentTermOfUseUrl',
      }),
    );
  });
});
