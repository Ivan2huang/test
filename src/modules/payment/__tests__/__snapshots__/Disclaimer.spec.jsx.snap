// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disclaimer Component should match to snapshot 1`] = `
<div>
  <div
    class="makeStyles-header-1"
    font-weight="bold"
    type="style-8"
  >
    Disclaimer
  </div>
  <div
    type="style-8"
  >
    <span>
      By purchasing, using or obtaining any products, services or consultations made available through HealthPass (a) you acknowledge and understand that HealthPass is developed, operated and maintained jointly by various third party providers; and (b) you agree that you have read and agree to the 
      <div
        class="makeStyles-link-2"
        color="outline"
        component="span"
        data-testid="open-term-of-use"
        type="style-8"
      >
        Term of Use
      </div>
       of HealthPass ("Terms of Use") governing your usage of HealthPass, which provide amongst other matters, that all products, services and consultations made available through HealthPass are not provided by OCBC but by Third Party Providers (as defined under the Terms of Use) under an independent contract entered between you and the Third Party Provider, for which OCBC has no responsibility.
    </span>
  </div>
</div>
`;
